lib/seattleflu/id3c/cli/command/etl/redcap_det_fh_airs.py

```diff
@@ -0,0 +1,910 @@
+    redcap_record_instances: List[REDCapRecord]) -> Optional[dict]:
+
+    if redcap_record_instances is None or len(redcap_record_instances) == 0:
+        LOG.warning(f"There are no record instances. Skipping record.")
+        return None
+
+    enrollments = [record for record in redcap_record_instances if record.event_name == ENROLLMENT_EVENT_NAME]
+
+    if not len(enrollments) == 1:
+        LOG.warning(f"There are {len(enrollments)} enrollment instances for record: {redcap_record_instances[0].get('subject_id')}. Skipping record.")
+        return None
+
+    enrollment = enrollments[0]
+
+    incomplete_enrollment_instruments = {
+                instrument
+                    for instrument
+                    in REQUIRED_ENROLLMENT_INSTRUMENTS
+                    if not is_complete(instrument, enrollment)
+            }
+
+    if incomplete_enrollment_instruments:
+        LOG.debug(f"The following required enrollment instruments «{incomplete_enrollment_instruments}» are not yet marked complete.")
+        return None
+
+    # If the participant's age < 18 do not continue
+    if (enrollment['scr_age'] == "" or int(enrollment['scr_age']) < 18):
+        LOG.debug("The participant age is blank or marked as under 18. Skipping record.")
+        return None
+
+    patient_entry, patient_reference = airs_build_patient(enrollment)
+    
+    if not patient_entry:
+        LOG.warning(f"Skipping record {enrollment.get('subject_id')} with insufficient information to construct patient")
+        return None
+
+    birthdate = parse_date_from_string(enrollment.get('enr_dob'))
+    if not birthdate:
+        LOG.warning(f"Record {enrollment.get('subject_id')} has an invalid or missing `enr_dob` value")
+
+    location_resource_entries = airs_build_location_info(db, cache, enrollment)
+
+    persisted_resource_entries = [patient_entry, *location_resource_entries]
+
+    for redcap_record_instance in redcap_record_instances:
+
+        event_type = None
+        collection_method = None
+
+        if redcap_record_instance.event_name == ENROLLMENT_EVENT_NAME:
+            event_type = EventType.ENROLLMENT
+
+        # TODO: I believe there's only one collection_method: swab_and_send
+        # The study documents say: 
+        #    If they report symptoms on a survey, 
+        #    they will be prompted to self-collect a nasal swab,
+        #    which will be delivered and returned via courier, 
+        #    and complete a brief symptom survey
+        # which doesn't even rhyme.
+        elif redcap_record_instance.event_name in ENCOUNTER_EVENT_NAMES:
+            if is_complete('weekly', redcap_record_instance):
+            #TODO: Do we skip events when no kit was sent, or are they still relevant?
+            #and \
+            #    (is_complete('airs_kit_activation_complete', redcap_record_instance) or \
+            #     is_complete('airs_kit_activation_2_complete', redcap_record_instance)):
```

Comment:
Are we still waiting for an answer on this one?

Reply:
No, via Kayla's conversation with them we know that the study wants data for participants who haven't received kits.